### PR TITLE
add typedefs for RecordRTC package

### PR DESCRIPTION
**Pull request by k-yle**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## recordrtc ([<kbd>unpkg</kbd>](https://unpkg.com/browse/recordrtc@latest/))
was missing the following properties:
1. getFromDisk
1. writeToDisk
1. MRecordRTC
1. getTracks
1. getSeekableBlob

<details>
<summary>as well as these 12 other properties...</summary>
<p>isElectron, Storage, MediaStreamRecorder, StereoAudioRecorder, CanvasRecorder, WhammyRecorder, Whammy, DiskStorage, GifRecorder, MultiStreamRecorder, RecordRTCPromisesHandler, WebAssemblyRecorder</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d2419ca48533c4d289c8cff011dd556d161d40f6
</p>

File: types/recordrtc/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    frameRate?: number;
+
+    /** used by WebAssemblyRecorder */
+    bitrate?: number;
+
+    /** used by MultiStreamRecorder - to access HTMLCanvasElement */
+    elementClass?: string;
+}
+
+declare class RecordRTC {
```
Comment (sheetalkamat) [inline types/recordrtc/index.d.ts:89]: Moving types except class into namespace RecordRTC will let library users use these types if they need to so thats a better way to organize things.

File: types/recordrtc/recordrtc-tests.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        });
+    },
+    console.error,
+);
+
+const canvas = document.querySelector('canvas')!;
+
+const instance2 = new RecordRTC(canvas, {
+    type: 'canvas',
+});
```
Comment (sheetalkamat) [inline types/recordrtc/recordrtc-tests.ts:23]: Do you need to address https://github.com/DefinitelyTyped/DefinitelyTyped/pull/46333#issuecomment-663768359 where there are lot more missing properties. Let us know if thats something correct to add or can be ignored.
Thanks

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/recordrtc/index.d.ts (hunk 1 above)
Comment (k-yle) [reply types/recordrtc/index.d.ts:89]: Hi @sheetalkamat, I'm not entirely sure what you mean by that. Are you suggesting something like this?
```ts
namespace RecordRTC {
    type State = 'inactive' | 'recording';

    interface Disk {
        // ...
    }

    // ...
}

declare class RecordRTC {
    constructor();

    // ... 
}

export = RecordRTC;

```

File: types/recordrtc/recordrtc-tests.ts (hunk 2 above)
Comment (k-yle) [reply types/recordrtc/recordrtc-tests.ts:23]: Those are internal properties which are not documented. Hence why I didn't add them to the definitions.

**Review by rbuckton — APPROVED** [review]

**Comment by k-yle** [comment]
Ready to merge

**Pull request merged**